core/src/jsMain/kotlin/SharedRepeatableTask.kt

```diff
@@ -0,0 +1,46 @@
+package com.juul.kable
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.CoroutineStart.DEFAULT
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+
+/**
+ * A mechanism for launching and awaiting some shared job repeatedly.
+ *
+ * The job is launched by calling [launchAsync]. Subsequent calls to [launchAsync] will return the
+ * same (i.e. shared) job if it is still executing.
+ */
+internal class SharedRepeatableTask(
+    private val scope: CoroutineScope,
+    private val job: suspend () -> Unit
```

Comment:
nit: rename variable to be clearer that this isn't a reference to a Coroutine `Job`

```suggestion
    private val action: suspend () -> Unit
```

or...

```suggestion
    private val task: suspend () -> Unit
```